src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1923,25 +1924,35 @@ private void PerformSchedulingActions(IEnumerable<ScheduleResponse> responses)
                         break;
 
                     case ScheduleActionType.CreateNode:
-                        var newNodes = new List<NodeInfo>();
-
-                        for (int i = 0; i < response.NumberOfNodesToCreate; i++)
+                        var newNodes = new ConcurrentBag<NodeInfo>();
+                        GetNodeConfiguration();
```

Comment:
Why are you calling this without consuming its return value?

Reply:
`GetNodeConfiguration` calculates and returns _nodeConfiguration, but it caches it so that if it's already been calculated, it doesn't calculate it again. This makes sure that it's already been calculated.

Reply:
Sounds good. Can you add a quick comment? Or rename method to something like `GetAndCacheNodeConfiguration` to avoid the comment.